### ISSUE TITLE
Annotated `...` placeholders inferred as `Const(Ellipsis)` instead of `Uninferable`

### ISSUE THREAD

**Issue by emmanuel-ferdman** [pylint-dev/astroid#3054]
### Description
When a stub or typed module uses `attr: int = ...`, astroid infers the value as the literal `Const(Ellipsis)` object. It should yield `Uninferable` since `...` is just a placeholder. This causes false positives downstream - pylint thinks the actual runtime value is `...` and makes wrong conclusions from that.

### Example
```python
import astroid

mod = astroid.parse("""
class Foo:
    x: int = ...
""")

cls = mod.body[0]
ann = cls.body[0]
inferred = list(ann.value.infer())
print(inferred)            # [<Const.Ellipsis l.3 ...>]  - should be [Uninferable]
```

**Comment by Pierre-Sassoulas** [comment]
Closing as invalid because it's actually an ellipsis, so we need to infer an ellipsis. I think it's an XY problem, let's open a new issue for the root cause.
